chromium_src/components/signin/internal/identity_manager/primary_account_manager.h

```diff
@@ -0,0 +1,20 @@
+/* Copyright (c) 2020 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_CHROMIUM_SRC_COMPONENTS_SIGNIN_INTERNAL_IDENTITY_MANAGER_PRIMARY_ACCOUNT_MANAGER_H_
+#define BRAVE_CHROMIUM_SRC_COMPONENTS_SIGNIN_INTERNAL_IDENTITY_MANAGER_PRIMARY_ACCOUNT_MANAGER_H_
+
+#define UpdatePrimaryAccountInfo                    \
```

Comment:
I think we can just do
```
#define RevokeSyncConsent                                   \
  RevokeSyncConsent_ChromiumImpl(                           \
      signin_metrics::ProfileSignout signout_source_metric, \
      signin_metrics::SignoutDelete signout_delete_metric); \
  void RevokeSyncConsent(                                   \
      signin_metrics::ProfileSignout signout_source_metric, \
      signin_metrics::SignoutDelete signout_delete_metric) {}
```
to stub out this method without changing the callers

Reply:
this doesn't work

In PR description I mentioned that both classes `PrimaryAccountMutatorImpl` and `PrimaryAccountManager` have method `RevokeSyncConsent`, so doing at `src/brave/chromium_src/components/signin/internal/identity_manager/primary_account_mutator_impl.cc`
```
#define RevokeSyncConsent RevokeSyncConsent_ChromiumImpl
#include "../../../../../../components/signin/internal/identity_manager/primary_account_mutator_impl.cc"
#undef RevokeSyncConsent
```
causes 


here is what happens
```

In file included from ../../brave/components/signin/public/identity_manager/brave_identity_manager.cc:6:
In file included from ../../brave/components/signin/public/identity_manager/brave_identity_manager.h:11:
In file included from ../../brave/chromium_src/components/signin/public/identity_manager/identity_manager.h:11:
In file included from ../../brave/chromium_src/components/signin/public/identity_manager/../../../../../../components/signin/public/identity_manager/identity_manager.h:17:
In file included from ../../brave/chromium_src/components/signin/internal/identity_manager/primary_account_manager.h:24:
../../brave/chromium_src/components/signin/internal/identity_manager/../../../../../../components/signin/internal/identity_manager/primary_account_manager.h:132:57: error: expected ')'
  void RevokeSyncConsent(signin_metrics::ProfileSignout signout_source_metric,
                                                        ^
../../brave/chromium_src/components/signin/internal/identity_manager/../../../../../../components/signin/internal/identity_manager/primary_account_manager.h:132:25: note: to match this '('
  void RevokeSyncConsent(signin_metrics::ProfileSignout signout_source_metric,
                        ^
../../brave/chromium_src/components/signin/internal/identity_manager/../../../../../../components/signin/internal/identity_manager/primary_account_manager.h:132:42: error: non-friend class member 'ProfileSignout' cannot have a qualified name
  void RevokeSyncConsent(signin_metrics::ProfileSignout signout_source_metric,
                         ~~~~~~~~~~~~~~~~^
../../brave/chromium_src/components/signin/internal/identity_manager/../../../../../../components/signin/internal/identity_manager/primary_account_manager.h:132:42: error: C++ requires a type specifier for all declarations
3 errors generated.

```
^ I don't understand exactly, but it looks like macro overlapping

Reply:
sorry I didn't specify detail enough. So this is the final patch
```patch
diff --git a/chromium_src/components/signin/internal/identity_manager/primary_account_manager.cc b/chromium_src/components/signin/internal/identity_manager/primary_account_manager.cc
new file mode 100644
index 0000000000..9695b8d40c
--- /dev/null
+++ b/chromium_src/components/signin/internal/identity_manager/primary_account_manager.cc
@@ -0,0 +1,10 @@
+
+#include "components/signin/internal/identity_manager/primary_account_manager.h"
+
+#define RevokeSyncConsent RevokeSyncConsent_ChromiumImpl
+#include "../../../../../../components/signin/internal/identity_manager/primary_account_manager.cc"
+#undef RevokeSyncConsent
+
+void PrimaryAccountManager::RevokeSyncConsent(
+    signin_metrics::ProfileSignout source_metric,
+    signin_metrics::SignoutDelete delete_metric) {}
diff --git a/chromium_src/components/signin/internal/identity_manager/primary_account_manager.h b/chromium_src/components/signin/internal/identity_manager/primary_account_manager.h
new file mode 100644
index 0000000000..324ba976f1
--- /dev/null
+++ b/chromium_src/components/signin/internal/identity_manager/primary_account_manager.h
@@ -0,0 +1,15 @@
+#ifndef BRAVE_CHROMIUM_SRC_COMPONENTS_SIGNIN_INTERNAL_IDENTITY_MANAGER_PRIMARY_ACCOUNT_MANAGER_H_
+#define BRAVE_CHROMIUM_SRC_COMPONENTS_SIGNIN_INTERNAL_IDENTITY_MANAGER_PRIMARY_ACCOUNT_MANAGER_H_
+
+#define RevokeSyncConsent                                   \
+  RevokeSyncConsent_ChromiumImpl(                           \
+      signin_metrics::ProfileSignout signout_source_metric, \
+      signin_metrics::SignoutDelete signout_delete_metric); \
+  void RevokeSyncConsent
+
+#include "../../../../../../components/signin/internal/identity_manager/primary_account_manager.h"
+
+#undef RevokeSyncConsent
+
+#endif  // BRAVE_CHROMIUM_SRC_COMPONENTS_SIGNIN_INTERNAL_IDENTITY_MANAGER_PRIMARY_ACCOUNT_MANAGER_H_
+
diff --git a/chromium_src/components/signin/internal/identity_manager/primary_account_mutator_impl.cc b/chromium_src/components/signin/internal/identity_manager/primary_account_mutator_impl.cc
new file mode 100644
index 0000000000..7b2851ed07
--- /dev/null
+++ b/chromium_src/components/signin/internal/identity_manager/primary_account_mutator_impl.cc
@@ -0,0 +1,14 @@
+#include "components/signin/internal/identity_manager/primary_account_mutator_impl.h"
+#include "components/signin/internal/identity_manager/primary_account_manager.h"
+
+#define RevokeSyncConsent RevokeSyncConsent_ChromiumImpl
+#include "../../../../../../components/signin/internal/identity_manager/primary_account_mutator_impl.cc"
+#undef RevokeSyncConsent
+
+namespace signin {
+
+void PrimaryAccountMutatorImpl::RevokeSyncConsent(
+    signin_metrics::ProfileSignout source_metric,
+    signin_metrics::SignoutDelete delete_metric) {}
+
+}  // namespace signin
diff --git a/chromium_src/components/signin/internal/identity_manager/primary_account_mutator_impl.h b/chromium_src/components/signin/internal/identity_manager/primary_account_mutator_impl.h
new file mode 100644
index 0000000000..50dc37cecb
--- /dev/null
+++ b/chromium_src/components/signin/internal/identity_manager/primary_account_mutator_impl.h
@@ -0,0 +1,16 @@
+#ifndef BRAVE_CHROMIUM_SRC_COMPONENTS_SIGNIN_INTERNAL_IDENTITY_MANAGER_PRIMARY_ACCOUNT_MUTATOR_IMPL_H_
+#define BRAVE_CHROMIUM_SRC_COMPONENTS_SIGNIN_INTERNAL_IDENTITY_MANAGER_PRIMARY_ACCOUNT_MUTATOR_IMPL_H_
+
+#include "components/signin/internal/identity_manager/primary_account_manager.h"
+#include "components/signin/public/identity_manager/primary_account_mutator.h"
+
+#define RevokeSyncConsent                                                      \
+  RevokeSyncConsent_ChromiumImpl(signin_metrics::ProfileSignout source_metric, \
+                                 signin_metrics::SignoutDelete delete_metric); \
+  void RevokeSyncConsent
+
+#include "../../../../../../components/signin/internal/identity_manager/primary_account_mutator_impl.h"
+
+#undef RevokeSyncConsent
+
+#endif  // BRAVE_CHROMIUM_SRC_COMPONENTS_SIGNIN_INTERNAL_IDENTITY_MANAGER_PRIMARY_ACCOUNT_MUTATOR_IMPL_H_
```

Reply:
thanks, I see now where I could make better 

fixed, or just applied your patch :) at https://github.com/brave/brave-core/compare/154cdbbe90be570271a8d5a259e8834a7bfe7c3a..44812e29461cbc6448b0e3805e13e043ae6c5519